contrib/inventory_builder/inventory.py

```diff
@@ -82,22 +82,30 @@ class KubesprayInventory(object):
     def __init__(self, changed_hosts=None, config_file=None):
         self.config_file = config_file
         self.yaml_config = {}
-        if self.config_file:
+        loadPreviousConfig = False
+        if self.config_file:  # Load previous YAML file
             try:
                 self.hosts_file = open(config_file, 'r')
-                self.yaml_config = yaml.load_all(self.hosts_file)
-            except OSError:
-                pass
-
+                self.yaml_config = yaml.load(self.hosts_file)
+            except OSError as e:
+                # I am assuming we are catching "cannot open file" exceptions
+                print(e)
+                sys.exit(1)
```

Comment:
Is this mandatory for adding `add` option?

Reply:
Will the `add` work without it? Yes, in that case it is not mandatory. 

However, not having it, means that a user can specify incorrect config path, and reading the file might fail. In that case, the exception is just ignored, and `yaml_config` is not read. Meaning, a user expected the script to use a config file, but it silently failed, and user doesn't even know something went wrong.